common/transport/mqtt/src/main/java/org/thingsboard/server/transport/mqtt/session/AbstractGatewaySessionHandler.java

```diff
@@ -452,148 +462,165 @@ public TransportProtos.PostTelemetryMsg postTelemetryMsgCreated(TransportProtos.
-
-                                @Override
-                                public void onFailure(Throwable t) {
-                                    log.debug("[{}][{}][{}] Failed to process device attributes command: [{}]", gateway.getTenantId(), gateway.getDeviceId(), sessionId, deviceName, t);
-                                }
-                            }, context.getExecutor());
-                });
-            } else {
+            if (CollectionUtils.isEmpty(attributesMsgList)) {
                 log.debug("[{}][{}][{}] Devices attributes keys list is empty", gateway.getTenantId(), gateway.getDeviceId(), sessionId);
                 throw new IllegalArgumentException("[" + sessionId + "] Devices attributes keys list is empty for [" + gateway.getDeviceId() + "]");
             }
+
+            attributesMsgList.forEach(attributesMsg -> {
+                String deviceName = checkDeviceName(attributesMsg.getDeviceName());
+                T deviceCtx = devices.get(deviceName);
+                if (deviceCtx != null) {
+                    processPostAttributesMsg(deviceCtx, attributesMsg.getMsg(), deviceName, msgId);
+                } else {
+                    Futures.addCallback(checkDeviceConnected(deviceName), new FutureCallback<>() {
+                        @Override
+                        public void onSuccess(@Nullable T deviceCtx) {
+                            processPostAttributesMsg(deviceCtx, attributesMsg.getMsg(), deviceName, msgId);
+                        }
+
+                        @Override
+                        public void onFailure(Throwable t) {
+                            log.debug("[{}][{}][{}] Failed to process device attributes command: [{}]", gateway.getTenantId(), gateway.getDeviceId(), sessionId, deviceName, t);
+                        }
+                    }, context.getExecutor());
+                }
+            });
         } catch (RuntimeException | InvalidProtocolBufferException e) {
             throw new AdaptorException(e);
         }
     }
 
-    protected void processPostAttributesMsg(MqttDeviceAwareSessionContext deviceCtx, TransportProtos.PostAttributeMsg postAttributeMsg, String deviceName, int msgId) {
-        transportService.process(deviceCtx.getSessionInfo(), postAttributeMsg, getPubAckCallback(channel, deviceName, msgId, postAttributeMsg));
+    protected void processPostAttributesMsg(MqttDeviceAwareSessionContext deviceCtx, JsonElement msg, String deviceName, int msgId) {
+        try {
+            TransportProtos.PostAttributeMsg postAttributeMsg = JsonConverter.convertToAttributesProto(msg.getAsJsonObject());
+            transportService.process(deviceCtx.getSessionInfo(), postAttributeMsg, getPubAckCallback(channel, deviceName, msgId, postAttributeMsg));
+        } catch (Throwable e) {
+            log.warn("[{}][{}][{}] Failed to process device attributes command: [{}]", gateway.getTenantId(), gateway.getDeviceId(), deviceName, msg, e);
```

Comment:
How will the client on the remote end get an error if some conversion does not go well? Logs are for the ThingsBoard sysadmin purposes only.